### ISSUE TITLE
PIL Image Ignores Orientation in EXIF Data

### ISSUE THREAD

**Issue by hmandsager** [scrapy/scrapy#6525]
### Description

This problem seems to exist only for photos taken with a Samsung device but when the image is loaded in PIL it doesn't properly handle the orientation in the EXIF data. 

### Steps to Reproduce

This section of code replicates (roughly) what happens in [Image Pipeline](https://github.com/scrapy/scrapy/blob/d2156696c45e023479ae1bdee8623bb6212e975c/scrapy/pipelines/images.py#L181)

```python
img_url = "https://auctioneersoftware.s3.amazonaws.com/mar/2024/10/ZIC8mPdtm32SMiJmkHcb5-jo.jpg"
response = requests.get(img_url)
img = Image.open(BytesIO(response.content))

img.show() # image will be upside down
```

```python
img_url = "https://auctioneersoftware.s3.amazonaws.com/mar/2024/10/ZIC8mPdtm32SMiJmkHcb5-jo.jpg"
response = requests.get(img_url)
img = Image.open(BytesIO(response.content))

exif = { ExifTags.TAGS[k]: v for k, v in img._getexif().items() if k in ExifTags.TAGS and ExifTags.TAGS[k] != 'MakerNote' }

if exif["Orientation"] == 3:
    # Rotate 180 degrees
    img = img.rotate(180, expand=True)
elif exif["Orientation"] == 6:
    # Rotate 90 degrees
    img = img.rotate(270, expand=True)
elif exif["Orientation"] == 8:
    # Rotate 270 degrees
    img = img.rotate(90, expand=True)

img.show() # image will show correct
```

**Expected behavior:** [What you expect to happen]
I would expect images saved through pipeline to match orientation of the site it was scraped from.

**Actual behavior:** [What actually happens]
Images are upside down

**Reproduces how often:** [What percentage of the time does it reproduce?]
Every time the orientation is given in this form it fails.

### Versions

```
Scrapy       : 2.11.1
lxml         : 5.2.1.0
libxml2      : 2.12.6
cssselect    : 1.2.0
parsel       : 1.9.1
w3lib        : 2.1.2
Twisted      : 24.3.0
Python       : 3.12.2 (main, Apr  5 2024, 14:54:25) [Clang 14.0.0 (clang-1400.0.29.202)]
pyOpenSSL    : 24.1.0 (OpenSSL 3.2.1 30 Jan 2024)
cryptography : 42.0.5
Platform     : macOS-14.4.1-arm64-arm-64bit
```

### Additional context

I believe that the `if/else` above would be the correct fix. Adding this either to `get_images` or `convert_images` functions make sense to me but interested in feedback.

**Comment by Gallaecio** [comment]
> This problem seems to exist only for photos taken with a Samsung device

I feel like this is not something to be fixed in Scrapy. I would report it to https://github.com/python-pillow/Pillow/issues instead, but I am not if this is something they should fix either, to be honest.

**Comment by wRAR** [comment]
Yeah, that part is unclear to me as I don't know how Pillow processes the orientation data. Is it processed automatically for some images and only skipped for others? If it's skipped for all of them then it makes sense to rotate the images when we strip the EXIF data but I would expect Pillow to have something ready for that (and Google suggests that it has).

**Comment by adam133** [comment]
Posted the equivalent issue in the Pillow repo - got this response. Maybe needs to be added to the pipeline code for scrapy? https://github.com/python-pillow/Pillow/issues/9109#issuecomment-3121097748

**Comment by radarhere** [comment]
I've created https://github.com/scrapy/scrapy/pull/6975 to try adding it to the pipeline.
